### ISSUE TITLE
FSM object as event emitter

### ISSUE THREAD

**Issue by vstirbu** [vstirbu/fsm-as-promised#7]
With this feature the FSM object is able to emit events on state changes when the provided target object is an instance of EventEmitter, or by default when no target is provided:

```javascript
      var fsm = StateMachine({
        initial: 'here',
        events: [
          { name: 'jump', from: 'here', to: 'there' },
          { name: 'walk', from: 'there', to: 'here' }
        ]
      });
      
      fsm.on('state', function (newState) {
        // do something when newState has some value
      });
```

The rationale behind this feature is that it enables the fsm state to be observed without hooking into the promise chain.

A prototype implementation is available in the the [emitter](https://github.com/vstirbu/fsm-as-promised/tree/emitter) branch.

cc: @boneskull

**Comment by boneskull** [comment]
This is *in addition to* the current implementation, not *instead of*, correct?

**Comment by vstirbu** [comment]
Yes! Sorry for the confusion :)

**Comment by boneskull** [comment]
I haven't looked at the branch, but I'd imagine you'd want to pass `oldState` into the event handler as well

**Comment by vstirbu** [comment]
solved by a23f345a252fbaf190fe84880723a9bf437641c3
